File: scripts/parse-feed.ts
import fs from "fs";
import path from "path";

// --- Läs CSV ---
const csvPath = path.join(process.cwd(), "datafeed_2824554.csv");
const raw = fs.readFileSync(csvPath, "utf-8");
const lines = raw.split("\n");
const headers = lines[0].split(",").map(h => h.trim().replace(/"/g, ""));

function parseRow(line: string): Record<string, string> {
  const values: string[] = [];
  let cur = "", inQ = false;
  for (const ch of line) {
    if (ch === '"') { inQ = !inQ; continue; }
    if (ch === "," && !inQ) { values.push(cur.trim()); cur = ""; }
    else cur += ch;
  }
  values.push(cur.trim());
  return Object.fromEntries(headers.map((h, i) => [h, values[i] ?? ""]));
}

// --- Kategorimappning ---
function mapCategory(merchantCat: string): string | null {
  const c = merchantCat.toLowerCase();
  if (c.includes("skor") || c.includes("sneaker") || c.includes("boots") || c.includes("loafer") || c.includes("sandal")) return "shoes";
  if (c.includes("jacka") || c.includes("kappa") || c.includes("coat") || c.includes("jacket") || c.includes("ytterkläder")) return "outerwear";
  if (c.includes("byxor") || c.includes("jeans") || c.includes("shorts") || c.includes("chino") || c.includes("slack")) return "bottom";
  if (c.includes("tröja") || c.includes("skjorta") || c.includes("t-shirt") || c.includes("hoodie") || c.includes("topp") || c.includes("stickad") || c.includes("sweatshirt") || c.includes("pikétröja") || c.includes("polotröja")) return "top";
  return null;
}

// --- Stilmappning ---
function mapStyle(name: string, cat: string): string[] {
  const n = name.toLowerCase();
  if (n.includes("slim") || n.includes("skinny")) return ["Minimalistisk", "Klassisk"];
  if (n.includes("cargo") || n.includes("wide") || n.includes("oversized")) return ["Streetwear", "Casual"];
  if (n.includes("linen") || n.includes("lin")) return ["Minimalistisk", "Skandinavisk"];
  if (cat === "jacket") return ["Casual", "Klassisk"];
  if (cat === "shoes") return ["Streetwear", "Casual"];
  return ["Casual"];
}

// --- Filtrera och deduplicera ---
const SKIP_CATS = ["underkläder", "strumpor", "kalsonger", "boxer", "badkläder", "pyjamas", "solglasögon", "handväskor", "parfym"];

const seen = new Set<string>();
const products: object[] = [];

for (let i = 1; i < lines.length; i++) {
  if (!lines[i].trim()) continue;
  const row = parseRow(lines[i]);

  const name = row["product_name"] || "";
  const price = parseFloat(row["search_price"]) || 0;
  const oldPrice = parseFloat(row["product_price_old"]?.split(" ")[0]) || 0;
  const imageUrl = row["merchant_image_url"] || "";
  const link = row["aw_deep_link"] || "";
  const colour = row["colour"] || "";
  const merchantCat = (row["merchant_category"] || "").toLowerCase();
  const inStock = row["in_stock"] === "1" || row["in_stock"] === "yes";
  const parentId = row["parent_product_id"] || row["aw_product_id"] || "";
  const brandRaw = row["brand_name"] || "NLY Man";

  // Hoppa över om ej i lager
  if (!inStock) continue;

  // Hoppa över underkläder etc
  if (SKIP_CATS.some(s => merchantCat.includes(s))) continue;
  if (SKIP_CATS.some(s => name.toLowerCase().includes(s))) continue;

  // Kategorimappning sker FÖRE seen-check så att storlekvarianter med
  // generisk kategori inte blockerar varianter med specifik kategori
  const category = mapCategory(merchantCat);
  if (!category) continue;

  // En produkt per parent_id (undvik storleksdubbletter)
  if (seen.has(parentId)) continue;
  seen.add(parentId);

  const brand = brandRaw
    .split(" ")
    .map(w => w.charAt(0).toUpperCase() + w.slice(1).toLowerCase())
    .join(" ");

  const isOnSale = oldPrice > price;

  products.push({
    brand,
    name,
    price,
    ...(isOnSale && { originalPrice: oldPrice }),
    imageUrl,
    category,
    style: mapStyle(name, category),
    colors: colour ? [colour.toLowerCase()] : [""],
    gender: "herr",
    condition: "new",
    isSecondHand: false,
    link,
  });

  if (products.length >= 120) break;
}

// --- Skriv ut mock-products.ts ---
const output = `export const mockProducts = ${JSON.stringify(products, null, 2)};\n`;
fs.writeFileSync(path.join(process.cwd(), "lib", "mock-products.ts"), output);
console.log(`✅ Klar! ${products.length} produkter skrivna till lib/mock-products.ts`);

File: scripts/parse-feed.py
import csv, json, os

csv_path = os.path.join(os.getcwd(), "datafeed_2824554.csv")
out_path = os.path.join(os.getcwd(), "lib", "mock-products.ts")

SKIP_NAMES = ["strumpor","kalsonger","boxer","badkläder","pyjamas","parfym","flerpack","underkläder"]
SKIP_CATS  = ["underkläder","strumpor","solglasögon","handväskor","parfym"]

def cat(c):
    c = c.lower()
    if "skor" in c or "sneaker" in c or "boots" in c: return "shoes"
    if "ytterkläder" in c or "kappor och jackor" in c: return "outerwear"
    if "byxor" in c or "jeans" in c or "shorts" in c: return "bottom"
    if "skjortor och toppar" in c or "tröjor" in c or "stickade" in c: return "top"
    return None

def style(name, c):
    n = name.lower()
    if "slim" in n or "skinny" in n: return ["Minimalistisk","Klassisk"]
    if "cargo" in n or "oversized" in n or "loose" in n or "baggy" in n: return ["Streetwear","Casual"]
    if "linen" in n or " lin" in n: return ["Minimalistisk","Skandinavisk"]
    if c == "outerwear": return ["Casual","Klassisk"]
    if c == "shoes": return ["Streetwear","Casual"]
    return ["Casual"]

seen = set()
out = []

debug_count = 0
with open(csv_path, encoding="utf-8-sig") as f:
    for row in csv.DictReader(f):
        debug_count += 1
        if debug_count == 1:
            import sys
            print(f"LOOP RUNS. stock={repr(row.get('in_stock'))} cat={repr(row.get('merchant_category','')[:40])}", file=sys.stderr)
        name  = (row.get("product_name") or "").strip()
        mc    = (row.get("merchant_category") or "").strip()
        stock = (row.get("in_stock") or "0").strip()
        pid   = ((row.get("parent_product_id") or "") or (row.get("aw_product_id") or "")).strip()
        brand = (row.get("brand_name") or "NLY Man").strip()
        img   = (row.get("merchant_image_url") or "").strip()
        link  = (row.get("aw_deep_link") or "").strip()
        col   = (row.get("colour") or "").strip()
        price_s = (row.get("search_price") or "0").strip().replace(",",".")
        old_s   = (row.get("product_price_old") or "0").strip().split()[0].replace(",",".")

        if stock not in ("1","yes","true"): continue
        if any(s in mc.lower() for s in SKIP_CATS): continue
        if any(s in name.lower() for s in SKIP_NAMES): continue
        c = cat(mc)
        if not c: continue
        if pid in seen: continue
        seen.add(pid)
        try:
            p = float(price_s) if price_s else 0
            op = float(old_s) if old_s else 0
        except ValueError:
            continue
        if p <= 0: continue
        b = " ".join(w.capitalize() for w in brand.split())
        prod = {"brand":b,"name":name,"price":p,"imageUrl":img,"category":c,
                "style":style(name,c),"colors":[col.lower()] if col else [""],
                "gender":"herr","condition":"new","isSecondHand":False,"link":link}
        if op > p: prod["originalPrice"] = op
        out.append(prod)
        if len(out) >= 120: break

ts = 'import { Product } from "./firestore-setup";\n\nexport const mockProducts: Omit<Product, "id">[] = ' + json.dumps(out, indent=2, ensure_ascii=False) + ";\n"
open(out_path, "w", encoding="utf-8").write(ts)
print(f"Klar! {len(out)} produkter → lib/mock-products.ts")
